### ISSUE TITLE
Spurious "unnecessary isinstance" with Type[]

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#2707]
This code:
```python
from typing import Type

class A:
    pass

class B(A):
    pass

def f(a: A, cls: Type[A]) -> bool:
    return isinstance(a, cls)
```

Produces an error:
`Unnecessary isinstance call; "A" is always an instance of "A"`

But that is incorrect, because we could call `f(A(), B)` and the `isinstance()` would return False.

On pylance v2021.12.2

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.196, which I just published. It will also be included in the next release of pylance.
